external-crates/move/crates/move-package/src/resolution/dependency_graph.rs

```diff
@@ -369,8 +369,53 @@ impl<Progress: Write> DependencyGraphBuilder<Progress> {
         };
 
         // combine the subgraphs for the dependencies into a single graph for the root package
-        dep_graphs.extend(dev_dep_graphs);
-        dep_names.extend(dev_dep_names);
+        let mut dep_graphs = {
+            let mut m = always_dep_graphs;
+            m.extend(dev_dep_graphs);
+            m
+        };
+        let dep_names = {
+            let mut m = always_dep_names;
+            m.extend(dev_dep_names);
+            m
+        };
+
+        for (
+            &dep_id,
+            DependencyGraphInfo {
+                g,
+                mode: _,
+                is_override,
+                is_external: _,
+                version: _,
+            },
+        ) in &mut dep_graphs
+        {
+            g.prune_overridden_pkgs(dep_id, &always_overrides, &dev_overrides, *is_override);
+        }
+
+        let overrides = {
```

Comment:
I'm guessing you have this here either for parity, or to avoid any additional clones? Otherwise I'd kinda like to see this higher up/before we prune the graphs as this is an obvious issue that we could report before doing any of the additional work to prune the subgraphs.

Reply:
It is here to avoid clones, yeah. I need the separation of the two when going into pruning.